src/server/generic_family.cc

```diff
@@ -1341,21 +1370,19 @@ OpResult<void> GenericFamily::RenameGeneric(CmdArgList args, bool skip_exist_des
   if (transaction->GetUniqueShardCnt() == 1) {
     transaction->ReviveAutoJournal();  // Safe to use RENAME with single shard
     auto cb = [&](Transaction* t, EngineShard* shard) {
-      return OpRen(t->GetOpArgs(shard), key[0], key[1], skip_exist_dest);
+      return OpRen(t->GetOpArgs(shard), key[0], key[1], destination_should_not_exist);
     };
     OpResult<void> result = transaction->ScheduleSingleHopT(std::move(cb));
 
     return result;
   }
 
   unsigned shard_count = shard_set->size();
-  Renamer renamer{Shard(key[0], shard_count)};
 
-  // Phase 1 -> Fetch  keys from both shards.
-  // Phase 2 -> If everything is ok, clone the source object, delete the destination object, and
-  //            set its ptr to cloned one. we also copy the expiration data of the source key.
-  renamer.Find(transaction);
-  renamer.Finalize(transaction, skip_exist_dest);
+  Renamer renamer{key[0], key[1], shard_count};
+
+  renamer.Initialize(transaction);
+  renamer.Finalize(transaction, destination_should_not_exist);
```

Comment:
I would move the input args to Renamer constructor and have only one call to execute the rename op. you can split it in rename calss to have the 2 steps of execution but there is no real reason for the caller to have both Initialize and Finalize here